lib/twitter_cldr/utils/script_detector.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+module TwitterCldr
+  module Utils
+    class ScriptDetectionResult
+
+      attr_reader :scores
+
+      def initialize(scores)
+        @scores = scores
+      end
+
+      def best_guess
+        max_score = 0
+        max_script_name = nil
+
+        scores.each_pair do |script_name, score|
+          if score > max_score
+            max_score = score
+            max_script_name = script_name
+          end
+        end
+
+        max_script_name
+      end
+
+      def score_for(script_name)
+        scores[script_name]
+      end
+
+    end
+
+    class ScriptDetector
+
+      class << self
+
+        def detect_scripts(text)
+          length = text.length.to_f
+
+          ScriptDetectionResult.new(
+            scores_for(text).each_with_object({}) do |(script_name, count), ret|
+              ret[script_name] = count / length
+            end
+          )
+        end
+
+        protected
```

Comment:
Considering how weird `protected` is in Ruby, I'd suggest to stick with `private`, unless it doesn't work.